pyclesperanto_prototype/_tier10/_to_igraph.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+from .._tier0 import plugin_function
+from .._tier0 import Image
+
+def to_igraph(adjacency_matrix:Image, centroids:Image=None):
+    """
+    Converts a given adjacency matrix to a iGraph [1] graph data structure.
+
+    Note: the given centroids typically have one entry less than the adjacency matrix is wide, because
+    those matrices contain a first row and column representing background. When exporting the networkx graph,
+    that first column will be ignored.
+
+    Parameters
+    ----------
+    adjacency_matrix : Image
+        m*m touch-matrix, proximal-neighbor-matrix or n-nearest-neighbor-matrix
+    centroids : Image, optional
+        d*(m-1) matrix, position list of centroids
+
+    Returns
+    -------
+    iGraph graph
+
+    See Also
+    --------
+    ..[1] https://igraph.org/
+    """
+    try:
+        import igraph
+    except ImportError:
+        raise ImportError("igraph is not installed. Please refer to the documentation https://igraph.org/python/")
+
+    igraph_graph = igraph.Graph.Adjacency(np.asarray(adjacency_matrix)[1:,1:].tolist())
```

Comment:
This is okay but note that `Graph.Adjacency()` now handles NumPy arrays directly (it calls `.tolist()` on the matrix automatically). However, if the matrices are large, this operation can potentially be costly as you need to construct one list for each row of the matrix, so if this is the case, let me know and I'll check whether we could support NumPy arrays in a more efficient manner.

Reply:
We really should support numpy arrays for all sorts of dense matrix operations. There should be a direct way to convert them to a C igraph matrix, without going through a Python list of lists.

Reply:
Hi @ntamas and @szhorvat ,

I'm just playing with [np.nonzero](https://numpy.org/doc/stable/reference/generated/numpy.nonzero.html) and the GPU-counterpart in this project [spots_to_pointlist](https://clij.github.io/clij2-docs/reference_spotsToPointList), They produce lists of coordinates where the adjacency-matrix contains `1`. That should be more efficient to construct the graph from those lists, right?

Reply:
Yes, that should be the preferred way to do it. If you have a 2D NumPy array with 2 columns and N rows, that should do the trick.